feature/experimental/gribi/otg_tests/fib_failed_due_to_hw_res_exhaust_test/fib_failed_due_to_hw_res_exhaust_test.go

```diff
@@ -0,0 +1,479 @@
+	if txPackets != 0 {
+		lossPct = lostPackets * 100 / txPackets
+	} else {
+		t.Errorf("Traffic stats are not correct %v", recvMetric)
+	}
+	if wantLoss {
+		if lossPct < 100-tolerancePct {
+			t.Errorf("Traffic is expected to fail %s\n got %v, want 100%% failure", flowName, lossPct)
+		} else {
+			t.Logf("Traffic Loss Test Passed!")
+		}
+	} else {
+		if lossPct > tolerancePct {
+			t.Errorf("Traffic Loss Pct for Flow: %s\n got %v, want 0", flowName, lossPct)
+		} else {
+			t.Logf("Traffic Test Passed!")
+		}
+	}
+}
+
+func verifyBgpTelemetry(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	var nbrIP = []string{atePort1.IPv4}
+	t.Logf("Verifying BGP state.")
+	bgpPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+
+	for _, nbr := range nbrIP {
+		nbrPath := bgpPath.Neighbor(nbr)
+		// Get BGP adjacency state.
+		t.Logf("Waiting for BGP neighbor to establish...")
+		status, ok := gnmi.Watch(t, dut, nbrPath.SessionState().State(), time.Minute, func(val *ygnmi.Value[oc.E_Bgp_Neighbor_SessionState]) bool {
+			state, ok := val.Val()
+			return ok && state == oc.Bgp_Neighbor_SessionState_ESTABLISHED
+		}).Await(t)
+		if !ok {
+			t.Fatal("No BGP neighbor formed")
+		}
+		state, _ := status.Val()
+		t.Logf("BGP adjacency for %s: %v", nbr, state)
+		if want := oc.Bgp_Neighbor_SessionState_ESTABLISHED; state != want {
+			t.Errorf("BGP peer %s status got %d, want %d", nbr, state, want)
+		}
+	}
+}
+
+// configureDUT configures port1-2 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Helper()
+	d := gnmi.OC()
+
+	p1 := dut.Port(t, "port1")
+	p2 := dut.Port(t, "port2")
+
+	gnmi.Replace(t, dut, d.Interface(p1.Name()).Config(), dutPort1.NewOCInterface(p1.Name(), dut))
+	gnmi.Replace(t, dut, d.Interface(p2.Name()).Config(), dutPort2.NewOCInterface(p2.Name(), dut))
+}
+
+func injectBGPRoutes(t *testing.T, args *testArgs) {
+	t.Helper()
+
+	if _, ok := vendorSpecRoutecount[args.dut.Vendor()]; !ok {
+		t.Fatalf("Please provide BGP route count for vendor to maxout FIB %v in var vendorSpecRoutecount ", args.dut.Vendor())
+	}
+	bgpNeti1Bgp4PeerRoutes := args.otgBgpPeer.V4Routes().Add().SetName(atePort1.Name + ".BGP4.Route")
+	bgpNeti1Bgp4PeerRoutes.SetNextHopIpv4Address(args.otgIPv4Device.Address()).
+		SetNextHopAddressType(gosnappi.BgpV4RouteRangeNextHopAddressType.IPV4).
+		SetNextHopMode(gosnappi.BgpV4RouteRangeNextHopMode.MANUAL)
+	bgpNeti1Bgp4PeerRoutes.Addresses().Add().
+		SetAddress(advertisedRoutesv4).
+		SetPrefix(advertisedRoutesv4MaskLen).
+		SetCount(vendorSpecRoutecount[args.dut.Vendor()]).SetStep(2)
+
+	args.otg.PushConfig(t, args.otgConfig)
+	time.Sleep(30 * time.Second)
+	args.otg.StartProtocols(t)
+	time.Sleep(30 * time.Second)
+}
+
+// awaitTimeout calls a fluent client Await, adding a timeout to the context.
+func awaitTimeout(ctx context.Context, c *fluent.GRIBIClient, t testing.TB, timeout time.Duration) error {
```

Comment:
Go style requires testing.TB to be the second parameter after context (and the first parameter when there is no context parameter)

Reply:
updated.

Reply:
@greg-dennis please could you check and help with one more network to use in script to generate 2.5M routes . As of now IP address blocks which are provided in CONTRIBUTING.md does not support 2.5M scale.
There is another test we need similar scale and that needs 4 such IP address groups. . Thank you.